packages/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp

```diff
@@ -7,12 +7,27 @@ namespace react = facebook::react;
 
 namespace expo {
 
+/**
+ Borrows the props map from the source props and applies the update given in the raw props.
+ */
+std::unordered_map<std::string, folly::dynamic> propsMapFromProps(const ExpoViewProps &sourceProps, const react::RawProps &rawProps) {
+  // Move the contents of the source props map – the source props instance will not be used anymore.
+  std::unordered_map<std::string, folly::dynamic> propsMap = std::move(sourceProps.propsMap);
+
+  // Iterate over values in the raw props object.
+  // Note that it contains only updated props.
+  rawProps.iterateOverValues([&propsMap](react::RawPropsPropNameHash hash, const char *name, const react::RawValue &value) {
+    std::string propName(name);
+    propsMap[propName] = (folly::dynamic)value;
+  });
+
+  return propsMap;
+}
+
 ExpoViewProps::ExpoViewProps(const react::PropsParserContext &context,
                              const ExpoViewProps &sourceProps,
                              const react::RawProps &rawProps)
                              : ViewProps(context, sourceProps, rawProps),
-                               proxiedProperties(
-                                 facebook::react::convertRawProp(context, rawProps, "proxiedProperties", sourceProps.proxiedProperties, {})) {
-}
+                               propsMap(propsMapFromProps(sourceProps, rawProps)) {}
```

Comment:
![screenshot_2023-04-29_at_18 52 50](https://user-images.githubusercontent.com/1714764/235351933-892aaf8b-a755-4c20-bc1e-205667abd089.png)

Since this object is already temporary, doing a move would be redundant and slower

Reply:
yep, i was wrong for this, the move is not necessary because it's already temporary object.